### ISSUE TITLE
Symbol and token for the class of semigroups

### ISSUE THREAD

**Issue by benjub** [metamath/set.mm#3801]
The current symbol, token and label for the class of semigroups are SGrp. I think SmGrp would be less ambiguous: when I see "SmGrp", I want to read it at once "semigroup", whereas with "SGrp", we have a moment of hesitation with subgroup, simple group, simplicial group, supergroup...

Maybe we can leave labels as such, and modify only symbol and token. Can I go with that ? (especially @tirix @avekens)

**Comment by avekens** [comment]
I agree that `SGrp` can be misleading. As for me, however, I do not associate "Sm" with "Semi". So why not call it (at least the symbol `SemiGrp`? It is not used often in set.mm yet (40 times).

**Comment by tirix** [comment]
I agree with @avekens, `SemiGrp` would be even better.

It would make sense to also change `SRing` to `SemiRing` at the same time, and I could follow suit in my mathbox with `SLMod` to `SemiLMod`, for semimodules (modules over semirings).

**Comment by benjub** [comment]
> I agree with @avekens, `SemiGrp` would be even better.

It looks a bit verbose, and unbalanced with the prefix fully spelt out but not the root (why not `Semigroup`, then?). I think that `SmGrp`, or actually better, `Smgrp`, is a good compromise, and maintains uniformity with the other group-like structures, where the symbol is made out of (all) the consonants of the word (Grp, Mnd, Mgm).

> It would make sense to also change `SRing` to `SemiRing` at the same time, and I could follow suit in my mathbox with `SLMod` to `SemiLMod`, for semimodules (modules over semirings).

Isn't that what we called a `Rng`? (And I think the `L` is extraneous: we have said once and for all that our modules are left-modules when they were defined, and that is enough.)

**Comment by tirix** [comment]
> Isn't that what we called a Rng?

The Rng would be a ring without identity (e.g. even numbers), and the semiring is a ring without inverses (e.g. natural numbers).

**Comment by tirix** [comment]
**TLDR**; To your original point: Ok, I'd be fine with the `SmGrp`, and generally using `Sm` for `Semi`.

A word about naming: in the 80's or 90's there were reasons to limit the lengths of identifiers (80 character line length, 8.3 file names, etc.). Those limitations do not exist anymore and nowadays the general recommendation is to prefer readability and maintainability: all naming guidelines I found in a quick internet search recommended fully spelled-out names. There's even been a [study](https://www.researchgate.net/publication/4242245_Whats_in_a_Name_A_Study_of_Identifiers). I try to adhere to those recommendations when programming.

For Metamath, things are a bit different. There is a 80 character line-length limitation, as well as indenting rules which eat up a part of this line length, and the fact that we're working on a single monolithic file which length we have to monitor closely. Those points could be worked around but that would deserve dedicated discussions. There's also the simple fact that the common usage in `set.mm` is to use abbreviations. So for Metamath I think we have to go along the practice.

**Comment by benjub** [comment]
Ok. I'm going to do the change to `Smgrp` (it's a single word, so no reason to put a capital G).  If for instance we go to 99-character lines in set.mm in the future, then maybe we could spell out everything (but then, to be consistent, also for `Group`, `Magma`, `Monoid`, maybe `CommMonoid`, `AssAlgebra`, etc.). We should also keep the equality between token and symbol in these cases. But I think `Smgrp` is already hardly ambiguous (or someone could think of "small group", with enough bad faith?).

I'll let you do what you see fit for your mathbox. I'd go for `Rig` ("ring without **n**egatives"), since "semiring" is rather used IIRC for nonunital rings (or rng, "ring witout **i**dentity"): the multiplicative magma is only required to be a semigroup, not a monoid, hence the "semi". I'd also remove the "L" from "SmLMod" (I'd remove it from LMod and LVec too: most textbooks use Mod and say simply "module" except in the definition or when differentiating is important, for instance in some in some cases for bimodules).

BTW: I just noticed that deprecated versions use `SemiGrp` and `Magma` (see both in https://us.metamath.org/mpeuni/df-sgrOLD.html).

**edit:** and I just noticed that `NrmSGrp` stands for "normal subgroup", so it's a good thing to abandon `SGrp` for "semigroup" indeed.

**Comment by benjub** [comment]
@tirix regarding https://us.metamath.org/mpeuni/mmtopstr.html can you:
* update "SGrp" to "Smgrp" and "SemiGroup" to "Semigroup"
* remove the arrows from Rng to Grp and from Ring to Abel and replace them with an arrow from Rng to Abel ?
* what is the double arrow from "Neutral element" to "Set" ? Probably a glitch ?
* the blue arrow "mulGrp" should probably target "Monoid"
* Preset --> Proset
* For `*Ring`, is it possible to put the star as a normal character instead of a superscript ? (same with `*r`)

**Comment by tirix** [comment]
Yes, I'm opening a dedicated issue to track that.

**Comment by tirix** [comment]
Any more opinions besides @avekens's ?
